components/tag-list/tag-list-item-mixin.js

```diff
@@ -0,0 +1,86 @@
+import '../colors/colors.js';
+import { css, html } from 'lit-element/lit-element.js';
+import { labelStyles } from '../typography/styles.js';
+
+export const TagListItemMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			/**
+			 * @ignore
+			 */
+			role: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			tabindex: { type: Number, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [labelStyles, css`
+			:host {
+				display: grid;
+				outline: none;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.tag-list-item-content {
+				height: 1rem;
+				margin: auto;
+				min-width: 0;
+				overflow: hidden;
+				text-overflow: ellipsis;
+				white-space: nowrap;
+			}
+			.tag-list-item-container {
+				background-color: var(--d2l-color-regolith);
+				border-radius: 6px;
+				box-shadow: inset 0 0 0 1px var(--d2l-color-gypsum), 0 2px 4px rgba(0, 0, 0, 0.03);
+				box-sizing: border-box;
+				color: var(--d2l-color-ferrite);
+				cursor: pointer;
+				display: flex;
+				min-height: 30px;
```

Comment:
So the problem is the `user-profile-image` which is in `px`.  The new size `x-small` size will be `24px` with `3px` of padding top and bottom, so without this small `font-size`s result in tags of different sizes.  I brought up switching the `user-profile-image`s to `rem` as well with Jeff and Noah, I could raise that again if we think that's the better solution.

Reply:
Hmm, so is the behaviour you want that all items are as tall as the tallest item? Because if so, you could use `align-items: stretch` on the outer list that's applying flexbox to its items.

Reply:
It seems like that only takes affect for tags on the same line, like this:
![image](https://user-images.githubusercontent.com/13419300/159057622-ff9990af-0b43-4a87-b2b9-f44e045645b2.png)

Which doesn't look awful, but I was trying to avoid measuring the height of a tag on each line to figure out how many lines we have.  And in actuality, will we have different types of tags mixed together? Probably not.  But playing around it looks like I may have accidentally fixed this later on by adding a height of `1rem` to the inner content, which hides overflow.  Means that the profile image would overflow on small font sizes though, like this:
![image](https://user-images.githubusercontent.com/13419300/159058250-bb3f217e-2c0e-4b8e-9479-6487bf71929e.png)
(this is root size of `14px`)